### PR TITLE
Hammerfix

### PR DESCRIPTION
**Pull request by WolfieWaffle**

### CODE REVIEW

File: src/main/java/toolbox/common/items/tools/ItemATHammer.java [hunk 1]
```diff
@@ -283,7 +303,7 @@ public RayTraceResult rayTraceBlocks(World world, EntityPlayer player) {
 		BlockPos pos = rt.getBlockPos();
 		IBlockState state = world.getBlockState(pos);
 
-		if (state.getBlock().isToolEffective(toolClass, state) || canHarvestBlock(state, stack)) {
+		if (state.getBlock().isToolEffective(toolClass, state) && canHarvestBlock(state, stack)) {
```
Comment (DaedalusGame) [inline src/main/java/toolbox/common/items/tools/ItemATHammer.java]: ForgeHooks.canHarvestBlock

File: src/main/java/toolbox/common/items/tools/ItemATHammer.java [hunk 2]
```diff
@@ -209,11 +209,31 @@ public boolean isDamageable() {
 	@Override
 	public boolean canHarvestBlock(IBlockState state, ItemStack stack) {
 		Block block = state.getBlock();
-		if (block == Blocks.OBSIDIAN || block == Blocks.REDSTONE_ORE || block == Blocks.LIT_REDSTONE_ORE) {
+		if (block == Blocks.OBSIDIAN
```
Comment (DaedalusGame) [inline src/main/java/toolbox/common/items/tools/ItemATHammer.java]: Doesn't the superclass method do all this already, except for the exact harvestlevel check? Infact, in an ideal world, this method should only run if
a) empty hand (in which case it doesn't run as we don't have a tool)
b) block has no toolclass set, note that while that may be the case in vanilla, *any* change to the block's harvest level will change the toolclass
c) tool has a negative toollevel for the block's toolclass

Comment (WolfieWaffle) [reply src/main/java/toolbox/common/items/tools/ItemATHammer.java]: I thought it did but I was running into issues. I will look into it though, I don't know much about Minecraft's code.

File: src/main/java/toolbox/common/items/tools/ItemATHammer.java (hunk 1 above)
Comment (WolfieWaffle) [reply src/main/java/toolbox/common/items/tools/ItemATHammer.java]: I'll see if I can change it to that.

**Comment by WolfieWaffle** [comment]
Any more problems I need to fix?

**Pull request merged**